bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/XmlText.java

```diff
@@ -127,8 +127,9 @@ public boolean equals(Object obj) {
      */
     @Override
     public boolean equals(Object o, Set<ValuePair> visitedValues) {
-        if (o instanceof XmlText rhsXMLText) {
-            return this.getTextValue().equals(rhsXMLText.getTextValue());
+        if ((o instanceof XmlText) || ((o instanceof XmlSequence)
+                && ((XmlSequence) o).isSingleton() && ((XmlSequence) o).getItem(0).getNodeType() == XmlNodeType.TEXT)) {
```

Comment:
```suggestion
        if ((o instanceof XmlText) || ((o instanceof XmlSequence)
                && ((XmlSequence) o).isSingleton() && ((XmlSequence) o).getItem(0).getNodeType() == XmlNodeType.TEXT)) {
```

I think this should be define as a separate function and name it properly, otherwise the logic is not clear.
ex:-

Reply:
Got it. I'll fix that.